### PR TITLE
SDL2 auxiliary fixes

### PR DESCRIPTION
**Pull request by Bedzior**
Pitifully, the SDL1.2 -> SDL2 migration did not go as intended. In the meantime however I made a few discoveries and a significant upgrade to Travis build times.

I wish to update the image in Docker Hub to `:latest` tag only after I confirm everything is in working order.

### CODE REVIEW

**Comment by Bedzior** [comment]
image
Much faster, methinks

File: .travis.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ env:
   global:
     - _C=build
     - WORKDIR=/spelunky
-    - BUILD_IMAGE=bedzior/spelunky-psp
+    - BUILD_IMAGE=bedzior/spelunky-psp:sdl2
```
Comment (Bedzior) [inline .travis.yml:17]: This could be changed back to implicit `:latest` when confirmed

File: build.dockerfile [hunk 2]
```diff
@@ -1,45 +1,51 @@
...
+        subversion \
+        git \
+        tcl \
+        unzip \
+        wget \
+        libcurl4-openssl-dev \
+        libgpgme-dev \
+        libarchive-dev \
+        libssl-dev \
+    && apt-get clean
```
Comment (Bedzior) [inline build.dockerfile:36]: Pruning intermediate layer

File: build.dockerfile [hunk 3]
```diff
@@ -1,45 +1,51 @@
...
 
 RUN git clone https://github.com/pspdev/psptoolchain.git --depth 1
 
 WORKDIR psptoolchain
 
-RUN ./toolchain-sudo.sh && ./toolchain-sudo.sh clean
+RUN ./toolchain-sudo.sh && ./toolchain-sudo.sh clean && rm -rf build
```
Comment (Bedzior) [inline build.dockerfile:42]: This prunes an intermediate layer as well, saves ~1.5GB

File: build.dockerfile (hunk 2 above)
Comment (Bedzior) [inline build.dockerfile:36]: Pruning intermediate layer

File: .travis.yml (hunk 1 above)
Comment (Bedzior) [inline .travis.yml:17]: This could be changed back to implicit `:latest` when confirmed and tagged in Docker Hub

File: build.dockerfile [hunk 4]
```diff
@@ -1,45 +1,51 @@
...
+        libfreetype6-dev \
+        zlib1g-dev \
+        libtool \
+        libtool-bin \
+        subversion \
+        git \
+        tcl \
+        unzip \
+        wget \
+        libcurl4-openssl-dev \
```
Comment (Bedzior) [inline build.dockerfile:32]: This and 3 below are newest PSP toolchain additions

File: src/video/src/Context_PSP.cpp [hunk 5]
```diff
@@ -55,7 +55,6 @@ bool Video::setup_gl()
         return false;
     }
 
-    SDL_GL_LoadLibrary(nullptr);
```
Comment (Bedzior) [inline src/video/src/Context_PSP.cpp:58]: This turned out to be completely unnecessary, according to [the docs](https://wiki.libsdl.org/SDL_GL_LoadLibrary#Remarks):
```
This should be done after initializing the video driver, but before creating any OpenGL windows. If no OpenGL library is loaded, the default library will be loaded upon creation of the first OpenGL window. 
```

Comment (dbeef) [reply src/video/src/Context_PSP.cpp:58]: The docs you linked refer to SDL2.

Nevertheless, it is not needed.
https://www.libsdl.org/release/SDL-1.2.15/docs/html/sdlglloadlibrary.html

File: .travis.yml (hunk 1 above)
Comment (dbeef) [reply .travis.yml:17]: So is this going to be changed?

Comment (Bedzior) [reply .travis.yml:17]: I'll do this later and then this PR will be fully ready to be merged

File: src/video/src/Context_PSP.cpp (hunk 5 above)
Comment (Bedzior) [reply src/video/src/Context_PSP.cpp:58]: My bad, sorry, but that's what I meant.

File: .travis.yml (hunk 1 above)
Comment (Bedzior) [reply .travis.yml]: Done

**Pull request merged**